File: ui-selenium/src/main/java/org/aeautomation/pages/AccountCreatedPage.java
package org.aeautomation.pages;

import org.aeautomation.core.BasePage;
import org.openqa.selenium.By;

/**
 * Page Object representing the "ACCOUNT CREATED!" confirmation screen.
 * Handles header verification and navigation back to the home page.
 */
public class AccountCreatedPage extends BasePage {

    private final By accountCreatedHeader = By.xpath("//b[contains(text(),'Account Created!')]");
    private final By continueButton = By.xpath("//a[@data-qa='continue-button']");

    public static final String HEADER_TEXT = "ACCOUNT CREATED!";

    /**
     * Retrieves the text from the account creation confirmation header.
     *
     * @return Trimmed header text string (e.g., "ACCOUNT CREATED!")
     */
    public String getHeaderText() {
        return getText(accountCreatedHeader);
    }

    /**
     * Clicks the "Continue" button to complete the registration process.
     *
     * @return New instance of HomePage representing the authenticated session home view
     */
    public HomePage clickContinue() {
        clickWithJS(continueButton);
        handleGoogleVignetteAd();

        return new HomePage();
    }
}

File: ui-playwright/pages/AccountCreatedPage.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './BasePage';
import { HomePage } from './HomePage';

/**
 * Page Object representing the account creation confirmation page (`/account_created`).
 */
export class AccountCreatedPage extends BasePage {

    /** Expected header text displayed upon successful account registration. */
    public static readonly ACCOUNT_CREATED_HEADER_TEXT = 'ACCOUNT CREATED!';

    /** Heading locator for the account creation confirmation screen (`getByTestId`). */
    readonly accountCreatedHeading: Locator;

    /** Link/Button locator to proceed past the confirmation screen (`getByTestId`). */
    readonly continueButton: Locator;

    constructor(page: Page) {
        super(page);

        this.accountCreatedHeading = page.getByTestId('account-created');
        this.continueButton = page.getByTestId('continue-button');
    }

    /**
     * Clicks the 'Continue' button after successful account creation
     * and navigates back to the home page.
     * 
     * @returns Promise resolving to a new HomePage instance
     */
    async clickContinue(): Promise<HomePage> {
        await this.click(this.continueButton);
        return new HomePage(this.page);
    }
}